test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -1919,6 +1930,10 @@ protected Collection<Class<? extends Plugin>> nodePlugins() {
         return Collections.emptyList();
     }
 
+    protected Class<? extends Plugin> telemetryPlugin() {
```

Comment:
Should come from `nodePlugins()`, please remove

Reply:
The idea here is we have added 2 types of integration tests :
1. In plugins:telemetry-otel which have loaded otel plugin in the test class and performed validations on the spans emitted. 
2. We want all the test cases should verify all the spans emitted have basic validations performed like SpansAreProperlyClosed, SpanIDsAreUnique etc. So if later we add more span in indexing or search code paths, our tests will already perform these validations. Here we have included a telemetryPlugin() in OpenSearchIntegTestCase.java, which is our MockTelemetryPlugin by default. We cannot use OTel plugin here as server dont have OTel specific dependency (SDKs etc).

Introduced new telemetryPlugin() because individual test writers can override nodeplugins() in testcase itself. Also if I include MockTelemetryPlugin By default in OpenSearchIntegTestCase.java, I wont be able to load Otel plugin in testing case 1 test cases. That's why introduced a new method telemetryPlugin().

Reply:
> Introduced new telemetryPlugin() because individual test writers can override nodeplugins() in testcase itself.

That's exactly the purpose

>Also if I include MockTelemetryPlugin By default in OpenSearchIntegTestCase.java, I wont be able to load Otel plugin in testing case 1 test cases.

Why?

```
    protected Collection<Class<? extends Plugin>> nodePlugins() {
        return nodePlugins().remove(MockTelemetryPlugin.class).add(OtelPlugin.class);
    }
```

Reply:
The super.nodePlugins() do not have MockTelemetryPlugin.class and almost all integ tests written are overriding this directly instead of calling super.nodePlugins(). Hence the purpose here is to run MockTelemetryPlugin by default for all integ test cases so basic validations are performed on all span data and all code paths. For testing Otel telemetry plugin added this flag https://github.com/opensearch-project/OpenSearch/pull/8444/files#diff-7c5bca73c48639083e61e82873def5f5d8b2410c1816a817628bd742612c73caR2073. Also checked other integ test cases are following this way https://github.com/opensearch-project/OpenSearch/pull/8444/files#diff-7c5bca73c48639083e61e82873def5f5d8b2410c1816a817628bd742612c73caR2063

Reply:
Correct, as we figured out, the getMockPlugins() is the right please, thank you.

Reply:
Moved it in getMockPlugins().